client/src/components/settings/exam-token.tsx

```diff
@@ -0,0 +1,117 @@
+import React, { useState } from 'react';
+import { Button, Panel, Modal, Spacer } from '@freecodecamp/ui';
+import { useTranslation } from 'react-i18next';
+import { FullWidthRow } from '../helpers';
+import { generateExamToken } from '../../utils/ajax';
+
+function ExamToken(): JSX.Element {
+  const [examToken, setExamToken] = useState<string | null>(null);
+  const [examTokenError, setExamTokenError] = useState<string | null>(null);
+
+  const [recentlyGenerated, setRecentlyGenerated] = useState(false);
+  const [generatedExamToken, setGeneratedExamToken] = useState(false);
```

Comment:
```suggestion
```
We don't need this. Either we have an `examToken`, in which case we show the modal, or we don't, in which case we don't.